script.js

```diff
@@ -98,3 +101,41 @@ function getData(featuredImage, live, source) {
   popUpImage.innerHTML = result;
   linkLive.innerHTML = link;
 }
+
+// form validation
+const form = document.querySelector('.form-view');
+
+function validForm(e) {
+  e.preventDefault();
+  const regex = /^[a-z0-9._%+-]+@[a-z0-9.-]+\.[a-z]{2,}$/;
+  if (emailValue.value.match(regex)) {
+    form.submit();
+    return true;
+  }
+  errorMsg.innerHTML = 'please email is invalid';
+  return false;
+}
+
+document.querySelector('.form-view').addEventListener('submit', validForm);
+
+function storeData() {
+  const formData = {
+    name: nameValue.value,
+    email: emailValue.value,
+    message: msgValue.value,
+  };
+  localStorage.setItem('formdata', JSON.stringify(formData));
+}
+
+document.querySelector('.form-view').addEventListener('submit', storeData);
```

Comment:
- [ ] Also make sure the data in the storage is in sync as the user is typing in the input fields. means the data are not supposed to be saved only when the user submits the form but when the user starts typing in the field.
Hint; You may consider using an `Input` event.